igc.cc

```diff
@@ -400,12 +467,72 @@ void IgcFormat::read()
       }
       break;
 
-      // These record types are discarded
+    case rec_fix_defn:
+    { // We need to scope this, or the compiler complains "transfer of control bypasses initialization of:"
+      // Not sure exactly what that means... something something scoping and initialization
+      /*
+       * The first three characters define the number of extensions present.
+       * We don't particularly care about that. After that, every group of seven
+       * bytes is 4 digits followed by three letters, specifying start end end
+       * bytes of each extension, and the kind of extension (always three chars)
+       *
+       * First, construct a hash of all available extensions.
+       * It may be better to construct the map and parse the record at the same time.
+       * However, for now we are contructing the hash map first, then iterating through it.
+       * The values in this hash contain all the necessary information on various extensions.
+      */
+      // TODO: Return this hash map for later use
+      QHash<QString, short>::const_iterator i;
+      printf("Extension types:"); //DELETEME development
```

Comment:
or qualify, where n is some positive small integer.  global_opts.debug is set by -D switch on command line, debug_level is zero by default(no debug).
 if (global_opts.debug_level >= n) {